src/crimson/net/SocketConnection.cc

```diff
@@ -61,37 +62,50 @@ SocketConnection::get_messenger() const {
   return &messenger;
 }
 
-bool SocketConnection::is_connected()
+seastar::future<bool> SocketConnection::is_connected()
 {
-  return !send_ready.failed();
+  return seastar::smp::submit_to(shard_id(), [this] {
+      return !send_ready.failed();
+    });
 }
 
 seastar::future<> SocketConnection::send(MessageRef msg)
 {
-  if (state == state_t::closing)
-    return seastar::now();
-  return seastar::with_gate(pending_dispatch, [this, msg=std::move(msg)] {
-      return do_send(std::move(msg))
-        .handle_exception([this] (std::exception_ptr eptr) {
-          logger().warn("{} send fault: {}", *this, eptr);
-          close();
+  return seastar::smp::submit_to(shard_id(), [this, msg=std::move(msg)] {
```

Comment:
i see. so w.r.t. m:n mapping, we have to do perform the i/o on some **certain** core, otherwise different cores talking to the same peer OSD could racing with each other.